pkg/kv/kvserver/allocator/storepool/store_pool.go

```diff
@@ -558,26 +571,50 @@ func (sp *StorePool) statusString(nl NodeLivenessFunc) string {
 // storeGossipUpdate is the Gossip callback used to keep the StorePool up to date.
 func (sp *StorePool) storeGossipUpdate(_ string, content roachpb.Value) {
 	var storeDesc roachpb.StoreDescriptor
-	// We keep copies of the capacity and storeID to pass into the
-	// capacityChanged callback.
-	var oldCapacity, curCapacity roachpb.StoreCapacity
-	var storeID roachpb.StoreID
 
 	if err := content.GetProto(&storeDesc); err != nil {
 		ctx := sp.AnnotateCtx(context.TODO())
 		log.Errorf(ctx, "%v", err)
 		return
 	}
-	storeID = storeDesc.StoreID
-	curCapacity = storeDesc.Capacity
+
+	sp.storeDescriptorUpdate(storeDesc)
+}
+
+// storeDescriptorUpdate takes a store descriptor and updates the corresponding
+// details for the store in the storepool.
+func (sp *StorePool) storeDescriptorUpdate(storeDesc roachpb.StoreDescriptor) {
+	// We keep copies of the capacity and storeID to pass into the
+	// capacityChanged callback.
+	var oldCapacity roachpb.StoreCapacity
+	storeID := storeDesc.StoreID
+	curCapacity := storeDesc.Capacity
+
+	now := sp.clock.PhysicalTime()
 
 	sp.DetailsMu.Lock()
 	detail := sp.GetStoreDetailLocked(storeID)
+
+	// We update the store descriptor to reflect the maximum IO Overload values
+	// that have been seen in the past tracking interval, updating the trackers
+	// with the new values prior to querying. This means that the most recent
+	// value for  IOThreshold.*Threshold is always used, together with the
+	// tracked recent maximum IOThreshold.L0NumFiles/L0NumSubLevels to
+	// calculate the score.
+	detail.storeHealthTracker.NumL0FilesTracker.Record(
+		now, float64(curCapacity.IOThreshold.L0NumFiles))
+	detail.storeHealthTracker.NumL0SublevelsTracker.Record(
+		now, float64(curCapacity.IOThreshold.L0NumSubLevels))
+	maxL0NumFiles, _ := detail.storeHealthTracker.NumL0FilesTracker.Query(now)
+	maxL0NumSublevels, _ := detail.storeHealthTracker.NumL0SublevelsTracker.Query(now)
+	storeDesc.Capacity.IOThreshold.L0NumFiles = int64(maxL0NumFiles)
```

Comment:
Why update storeDesc.Capacity, and not curCapacity?

Reply:
Removed this. Instead the value is just taken as is. I talked to @andrewbaptist about this and the effort involved to refactor the allocation code wasn't worth the benefit the max windowing brings.